collector/collector.go

```diff
@@ -139,7 +140,18 @@ func (c *Collector) GetRawNetworkMetricsHandler(w http.ResponseWriter, req *http
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
-	if _, err := w.Write(batchBytes); err != nil {
+
+	writer, err := gzip.NewWriterLevel(w, gzip.BestCompression)
+	if err != nil {
+		c.log.Errorf("cannot create gzip writer: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	defer writer.Close()
+
+	w.Header().Add("Content-Encoding", "gzip")
```

Comment:
Should we check the Accept-Encoding header? To make compliant and safer transition?

Reply:
It is used by exporter which accepts gzipped content, but nonetheless good point. Will address it